mlir/include/mlir/Transforms/DialectConversion.h

```diff
@@ -898,6 +898,11 @@ class ConversionPatternRewriter final : public PatternRewriter {
   /// Replace the given operation with the new value ranges. The number of op
   /// results and value ranges must match. The given  operation is erased.
   void replaceOpWithMultiple(Operation *op, ArrayRef<ValueRange> newValues);
+  template <typename RangeT>
+  void replaceOpWithMultiple(Operation *op, RangeT &&newValues) {
```

Comment:
Ok, I know I pointed out the issue here cause `ArrayRef<ValueRange>` is a list of references but this is doing a copy under the hood. Can we make this `ArrayRef<SmallVector<T>>`. This way, no copy is required, with the caller holding the memory?

Reply:
To avoid the copy, an `ArrayRef<SmallVector<Value, 1>>` must be used because the internal dialect conversion representation in the mapping data structure is `SmallVector<Value, 1>`.

It's bit tricky to make all the overloads work. But I think I found a solution. @zero9178 Can you give this another sanity check? You know this stuff better...